test/hotspot/jtreg/compiler/codecache/CheckSegmentedCodeCache.java

```diff
@@ -58,7 +58,7 @@ private static void verifySegmentedCodeCache(ProcessBuilder pb, boolean enabled)
                 out.shouldContain(NON_METHOD);
             } catch (RuntimeException e) {
                 // Check if TieredCompilation is disabled (in a client VM)
-                if (!out.getOutput().contains("-XX:+TieredCompilation not supported in this VM")) {
+                if (Platform.isTieredSupported()) {
```

Comment:
That's probably not a complete set of conditions of when the segmented code cache is disable. But if it solves the current issue, I think that's fine.